### PR TITLE
:bug: Add hubHash in addon client cert secret as a workaround for issue-10093

### PR DESCRIPTION
**Pull request by xuezhaojun**
Fixes: https://issues.redhat.com/browse/ACM-10093

Note: this fix is only a workaround, and is not included in the main branch or backplane-2.6+.

### CODE REVIEW

File: pkg/registration/spoke/addon/registration_controller.go [hunk 1]
```diff
@@ -213,6 +215,16 @@ func (c *addOnRegistrationController) startRegistration(ctx context.Context, con
...
 	if config.registration.SignerName == certificatesv1.KubeAPIServerClientSignerName {
 		additonalSecretData[clientcert.KubeconfigFile] = c.kubeconfigData
+	} else {
+		// TODO: this is a workaround for issue-10093. it doesn't contains in the future MCE release(2.6+)
+		// Decode CA from kubeconfigData, hash it and store in the additionalSecretData
+		config, err := clientcmd.Load(c.kubeconfigData)
+		if err != nil {
+			klog.Error(err, "Failed to load kubeconfig, skip adding hubHash to additionalSecretData")
+		} else {
+			hash := sha256.Sum256(config.Clusters["default-cluster"].CertificateAuthorityData)
```
Comment (elgnay) [inline pkg/registration/spoke/addon/registration_controller.go:225]: The import-controller generates the bootstrap hub kubeconfig with cluster name `default-cluster`, but I'm not sure if there is any case where the bootstrap hub kubeconfig has a different cluster name. In that case, the line will fail.

Comment (xuezhaojun) [reply pkg/registration/spoke/addon/registration_controller.go]: Agree, updated to avoid potential name confictions.

**Comment by xuezhaojun** [comment]
@qiujian16  Request a bypass merge， the e2e error is about manifestwork status report and is not related with changes in this PR。 Will have a seperate PR to fix it：https://github.com/stolostron/ocm/pull/36

**Review by qiujian16 — COMMENTED** [review]
/approve

**Pull request merged**